Use orchestra's ninja on all components

mrjackv opened the pull request:
Fixes issue where cmake's hardcoded ninja path causes failed builds
~~Ideally should be merged after #112 since the second commit would be elided (but it's necessary on its own regardless)~~

mrjackv committed:
Use orchestra ninja

mrjackv committed:
python: add missing libraries

Add the libb2 and liblzma libraries that python required that were
incorrectly linked to the host system's libraries

aleclearmind commented on .orchestra/config/components/xz.yml:
```diff
@@ -0,0 +1,27 @@
+build_system: make
+build_dependencies:
+  - host-c-toolchain
+  - glibc
```
I think this is implied by `host-c-toolchain`, can you double check?

mrjackv replied:
Checked, it is indeed

aleclearmind commented on .orchestra/config/components/xz.yml:
```diff
@@ -0,0 +1,27 @@
+  extract.sh --into "$BUILD_DIR" (@= source_url @)
+
+  cd "$BUILD_DIR"
+  ./configure --enable-werror --prefix="$ORCHESTRA_ROOT"
```
Multiline, also usually for dependencies we don't check we don't enable `-Werror` it's just going to fire back at us if we bump the compiler and a new warning pops up.

mrjackv replied:
Done

aleclearmind commented on .orchestra/config/components/python.yml:
```diff
@@ -60,6 +60,8 @@ builds:
       - openssl
       - readline
       - sqlite
+      - libb2
+      - xz
```
F-, I wonder how is this possible? We should now be using `--sysroot` properly which should lead to failures if a dependency is missing in orchestra.

How did you detected these were missing?
Are you sure these are not optional deps?

This would need further investigation, but not today.

mrjackv replied:
> How did you detected these were missing?

I was looking at python's build log and noticing a lot of log spam complaining that the `_blake2` module was not present. It seems that while it's an optional dependency, python will complain about its absence every time the `hashlib` module is imported. After that I've looked at the build log a bit closer and it seems that there were a couple of other optional libraries that probably got linked to the OS in the past.
The two main ones are the ones added to the PR:
* libb2, since [python prefers this implementation](https://github.com/python/cpython/blob/main/Lib/hashlib.py#L73)
* liblzma, needed for the `lzma` module

As said above, both are optional (together with `libuuid` for `uuid` and `libtirpc` for `nis`), but IMHO their contribution justifies the inclusion (as opposed to the other two)

aleclearmind replied:
I'm fine with including them, I'd be sad if, despite the `--sysroot` efforts, we were still pulling libraries from the host.
But according to what you're saying this is not the case, so I'm happy.

mrjackv replied:
Yes, in fact, by virtue of the `--sysroot` efforts I noticed that python was pulling in libraries from outside

aleclearmind commented on .orchestra/ci/install-system-dependencies.sh:
```diff
@@ -50,7 +50,6 @@ PACKAGES+=(gettext)
 PACKAGES+=(libc-dev)
 PACKAGES+=(libtool)
 PACKAGES+=(m4)
-PACKAGES+=(ninja-build)
```
Nice

aleclearmind commented on .orchestra/config/lib/cmake.lib.yml:
```diff
@@ -114,8 +115,11 @@
 
-  #@ if/end build_dependencies:
-  build_dependencies: #@ build_dependencies
+  build_dependencies:
+  - cmake
+  #@ if/end build_system == "Ninja":
+  - ninja
```
Nice

mrjackv committed:
`host-{c,cxx}-toolchain` implies glibc

Remove redundant dependency on `glibc` when host-{c,cxx}-toolchain
component is used, which already declares it as a dependency

aleclearmind commented:
Merged in c5d8126. Thanks!
